lcls_tools/common/data_analysis/projection_fit/gaussian_model.py

```diff
@@ -0,0 +1,89 @@
+import numpy as np
+from scipy.stats import norm, gamma
+from scipy.ndimage import gaussian_filter
+from lcls_tools.common.data_analysis.projection_fit.method_base import MethodBase
+
+
+class GaussianModel(MethodBase):
+    """
+    GaussianModel Class that finds initial param values for gaussian distribution
+        and builds probability density functions for the likelyhood a param
+        to be that value based on those initial param values
+
+    - passing this class the variable profile_data automatically updates
+        the initial values and and probability density functions to match that data
+    """
+
+    param_names: list = ["amplitude", "mean", "sigma", "offset"]
+    param_bounds: np.ndarray = np.array(
+        [[0.01, 1.0], [0.01, 1.0], [0.01, 5.0], [0.01, 1.0]]
+    )
+
+    def __init__(self, profile_data: np.ndarray = None):
+        if profile_data is not None:
+            self.profile_data = profile_data
+            self.find_init_values(self.profile_data)
+            self.find_priors(self.profile_data)
+
+    def find_init_values(self, data: np.array) -> np.array:
+        offset = float(np.min(data))
+        amplitude = np.max(gaussian_filter(data, sigma=5)) - offset
+        mean = np.argmax(gaussian_filter(data, sigma=5)) / (len(data))
+        sigma = 0.1
+        self.init_values_list = np.array([amplitude, mean, sigma, offset])
+        self.init_values = {"amplitude":amplitude,"mean":mean,"sigma":sigma,"offset":offset}
+        # if use_priors = True in projection_fit then find priors? use case where projection fit is instantiated with use_priors = False then flag is changed but you have no priors
+        self.find_priors()
```

Comment:
I highly recommend calling self.find_priors when init_values are found.

Lets consider the workflow for projection_fit.fit_projection(some_projection_data).
1.) This method passes normalized projection data to projection.setup_model. 
2.) projection.setup_model calls the setter method for gaussian_model.profile_data
3.) The setter method for gaussian_model.profile_data triggers self.find_init_values
4.) those init values are used in projection_fit.fit_model()
5.) now consider the case when projection_fit.use_priors = True, every thing works smoothly right? Well no.
6.) projection_fit.use_priors = True can be set upon initialization or after. Also, the advantage of having gaussian_model.profile_data be a @property is that it can be updated easily. Meaning that we can call projection_fit.fit_projection( __with_any_1d_array__). 
7.) So use_priors = True but priors have not changed. Which is not good. Calling gauss_model.find_priors inside 
projection_fit.fit_model() is also bad.